### PR TITLE
Add Server-Sent Events BBE

### PR DESCRIPTION
**Pull request by MohamedSabthar**
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/6790

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: examples/http-sse-service/http_sse_service.bal [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+service /stocks on new http:Listener(9090) {
+    resource function get .() returns stream<http:SseEvent, error?> {
+        StockPriceEventGenerator generator = new;
+        return new (generator);
+    }
+}
+
+class StockPriceEventGenerator {
+    int i = 0;
```
Comment (ayeshLK) [inline examples/http-sse-service/http_sse_service.bal:13]: Shall we rename this variable to something meaningful like `resetCounter` ?

Comment (MohamedSabthar) [reply examples/http-sse-service/http_sse_service.bal]: updated with [bc20a34](https://github.com/ballerina-platform/ballerina-distribution/pull/5533/commits/bc20a340e973a2fe6bd0cfff03292ead9b80cfe5)

File: examples/index.json [hunk 2]
```diff
@@ -2327,6 +2335,14 @@
         "verifyOutput": false,
         "disablePlayground": false,
         "isLearnByExample": false
+      },
+      {
+        "name": "Server-Sent Events",
```
Comment (MaryamZi) [inline examples/index.json:2340]:
```suggestion
        "name": "Server-sent events",
```

Comment (MaryamZi) [reply examples/index.json:2340]: Fix wherever applicable.

File: examples/http-sse-client/http_sse_client.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
+# HTTP client - Server-Sent Events
+
+The `http:Client` in Ballerina supports receiving real-time data from services using Server-Sent Events (SSE). It allows payload-binding of a stream of `http:SseEvent` when consuming SSE from a service. This payload binding fails if the content type header is not present in the response or does not have the value `text/event-stream`.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/http-sse-client/http_sse_client.md:3]: Using the first one similar to other BBEs.

File: examples/http-sse-client/http_sse_client.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# HTTP client - Server-sent events
+
+The `http:Client` in Ballerina supports receiving real-time data from services using Server-Sent Events (SSE). It allows payload-binding of a stream of `http:SseEvent` when consuming SSE from a service. This payload binding fails if the content type header is not present in the response or does not have the value `text/event-stream`.
```
Comment (MaryamZi) [inline examples/http-sse-client/http_sse_client.md:3]: Server-Sent Events -> Server-sent events

Let's update wherever applicable.

File: examples/http-sse-client/http_sse_client.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+::: code http_sse_client.bal :::
+
+## Prerequisites
+- Run the HTTP service given in the [Server-sent events](/learn/by-example/http-sse-service/) example.
+
+Run the client program by executing the following command..
```
Comment (MaryamZi) [inline examples/http-sse-client/http_sse_client.md:10]:
```suggestion
Run the client program by executing the following command.
```

File: examples/http-sse-service/http_sse_service.bal
Comment (MaryamZi) [inline examples/http-sse-service/http_sse_service.bal:1]: Don't we need some comments?

File: examples/http-sse-service/http_sse_service.md [hunk 6]
```diff
@@ -0,0 +1,17 @@
+# HTTP service - Server-sent events
+
+The `http:Service` in Ballerina supports pushing real-time data to clients using Server-Sent Events (SSE). A stream of `http:SseEvent` can be returned from service resource methods. This feature automatically handles sending SSE and sets the content type to `text/event-stream`, and the transfer encoding is automatically set to chunked.
```
Comment (MaryamZi) [inline examples/http-sse-service/http_sse_service.md:3]:
```suggestion
Ballerina HTTP services support pushing real-time data to clients using Server-event events (SSE). A stream of type `http:SseEvent` can be returned from service resource methods. This feature automatically handles sending SSE and sets the content type to `text/event-stream` and the transfer encoding to chunked.
```

File: examples/http-sse-service/http_sse_service.bal
Comment (TharmiganK) [reply examples/http-sse-service/http_sse_service.bal:1]: Yes, we could say what that resource does(streaming stock prices) and add some comments regrading the stream generator class and its purpose

Comment (MohamedSabthar) [reply examples/http-sse-service/http_sse_service.bal:1]: Added https://github.com/ballerina-platform/ballerina-distribution/pull/5533/commits/faa16418bf1be9ac71c397183b6c5d9b5c968790

File: examples/http-sse-service/http_sse_service.bal [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    // This function returns a stream of SSE with stock prices
+    resource function get .() returns stream<http:SseEvent, error?> {
+        // Create a new instance of StockPriceEventGenerator to generate stock price events
+        StockPriceEventGenerator generator = new;
+        // Return a new stream that uses the generator to produce events
+        return new (generator);
+    }
+}
+
+// Define a class to generate stock price events
```
Comment (MaryamZi) [inline examples/http-sse-service/http_sse_service.bal:15]:
```suggestion
// Define a stream implementor that can be used to create a stream 
// of `http:SseEvent`, representing stock price events.
```

File: examples/http-sse-service/http_sse_service.bal [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+class StockPriceEventGenerator {
+    int eventCounter = 0;
+
+    public isolated function next() returns record {|http:SseEvent value;|}|error? {
+        // If the eventCounter reaches 5, stop generating events by returning nil
```
Comment (MaryamZi) [inline examples/http-sse-service/http_sse_service.bal:20]:
```suggestion
        // If the eventCounter reaches 5, stop generating events by returning nil.
```

File: examples/http-sse-client/http_sse_client.bal
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/http-sse-client/http_sse_client.bal:1]: There isn't much to add here, as the description section already explains the return type/payload binding.

File: examples/http-sse-client/http_sse_client.md [hunk 9]
```diff
@@ -0,0 +1,16 @@
+# HTTP client - Server-sent events
+
+The `http:Client` in Ballerina supports receiving real-time data from services using Server-sent events (SSE). It allows payload-binding of a stream of `http:SseEvent` when consuming SSE from a service. This payload binding fails if the content type header is not present in the response or does not have the value `text/event-stream`.
```
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/http-sse-client/http_sse_client.md:3]: +1, will update

File: examples/http-sse-client/http_sse_client.bal
[the remark this answers is not in the document]
Comment (MaryamZi) [reply examples/http-sse-client/http_sse_client.bal:1]: May help to have it in the code also for someone who just copies the code and tries it out. That way they wouldn't have to go back and forth between bio and VS Code for example.

File: examples/http-sse-client/http_sse_client.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/http-sse-client/http_sse_client.md]: updated

File: examples/http-sse-client/http_sse_client.bal
[the remark this answers is not in the document]
Comment (MohamedSabthar) [reply examples/http-sse-client/http_sse_client.bal:1]: Added https://github.com/ballerina-platform/ballerina-distribution/pull/5533/commits/54580a7cecf69fd4a85d3ec4711182387a8be3a0

File: examples/http-sse-service/http_sse_service.bal [hunk 10]
```diff
@@ -0,0 +1,33 @@
+import ballerina/http;
+import ballerina/lang.runtime;
+import ballerina/random;
+
+service /stocks on new http:Listener(9090) {
+    // This resource method returns a stream of `http:SseEvent` (with stock prices)
+    // to push real-time data to clients using server-event events (SSE).
```
Comment (MaryamZi) [inline examples/http-sse-service/http_sse_service.bal:7]:
```suggestion
    // to push real-time data to clients using server-sent events (SSE).
```

File: examples/http-sse-service/http_sse_service.md [hunk 11]
```diff
@@ -0,0 +1,17 @@
+# HTTP service - Server-sent events
+
+Ballerina HTTP services support pushing real-time data to clients using Server-event events (SSE). A stream of type `http:SseEvent` can be returned from service resource methods. This feature automatically handles sending SSE and sets the content type to `text/event-stream` and the transfer encoding to chunked.
```
Comment (MaryamZi) [inline examples/http-sse-service/http_sse_service.md:3]:
```suggestion
Ballerina HTTP services support pushing real-time data to clients using server-sent events (SSE). A stream of type `http:SseEvent` can be returned from service resource methods. This feature automatically handles sending SSE and sets the content type to `text/event-stream` and the transfer encoding to chunked.
```

**Review by MaryamZi — COMMENTED** [review]
The rest LGTM.

**Review by TharmiganK — APPROVED** [review]

**Pull request merged**